d_do_test: Enforce colon after test directives

MoonlightSentinel opened the pull request:
Because it should always indicate that it's a test parameter and not an unrelated token that contains it. For reference, the GDC test suite currently ignores test parameters without a colon.
    
Replaced the search for `GDB_SCRIPT` because shell scripts usually use  `GDB_SCRIPT=...` and the check doesn't require the actual content.

MoonlightSentinel committed:
Add missing colons after test directives

They were not required by the current test runner but will
be enforced by future patches

ibuclaw commented:
> Because it should always indicate that it's a test directive and not an unrelated token that contains it. This is (partially?) enforced within the GDC test suite and should also be checked here.

Actually, the GDC test suite ignores it, because the Dejagnu scripts only match the directive _with_ colon.

i.e:
```
if [regexp -- {DFLAGS\s*:\s*(.*)} $line match args] {
 ...
} elseif [regexp -- {DISABLED:} $copy_line] {
 ...
}
 ```

ibuclaw commented:
I'd prefer this to use words rather than symbols in the commit summary:
```
d_do_test: Enforce colon after test directives
```

MoonlightSentinel commented:
> Actually, the GDC test suite ignores it, because the Dejagnu scripts only match the directive _with_ colon.

Thanks, I've corrected the commit summary accordingly.

> I'd prefer this to use words rather than symbols in the commit summary:

Sure

WalterBright commented on test/fail_compilation/invalid_lib.d:
```diff
@@ -1,7 +1,7 @@
-REQUIRED_ARGS(linux freebsd osx openbsd) fail_compilation/extra-files/fake.a
-REQUIRED_ARGS(windows) -m32 fail_compilation/extra-files/fake.lib
+REQUIRED_ARGS(linux freebsd osx openbsd): fail_compilation/extra-files/fake.a
+REQUIRED_ARGS(windows): -m32 fail_compilation/extra-files/fake.lib
```
Where did the (...) syntax come from? This is not documented.

MoonlightSentinel replied:
Apparently missing from the README, will add in another PR.

WalterBright commented on test/tools/d_do_test.d:
```diff
@@ -318,6 +328,65 @@ bool findTestParameter(const ref EnvData envData, string file, string token, ref
+void main() {}
+
+// BAR(windows): all_win
+// BAR(windows64): 64_win
```
`windows64` is not documented anywhere

MoonlightSentinel replied:
Same syntax as `DISABLED` values.

WalterBright commented on test/fail_compilation/fail18417.d:
```diff
@@ -1,4 +1,4 @@
-// REQUIRED_ARGS : -de
+// REQUIRED_ARGS: -de
```
If the space before : is allowed or not is not documented anywhere.

MoonlightSentinel replied:
Fixed

WalterBright reviewed (dismissed):
It is not clear to me whether `XXTEST_OUTPUT:` is recognized as `TEST_OUTPUT` or not. (It shouldn't be.)

MoonlightSentinel committed:
d_do_test: Enforce colon after test / output parameters

Because it should always indicate that it's a test parameter and not
an unrelated token that contains it. For reference, the GDC test suite
currently ignores test parameters without a colon.

Replaced the search for `GDB_SCRIPT` because shell scripts usually use
`GDB_SCRIPT=...` and the check doesn't require the actual content.

MoonlightSentinel commented:
Adapted to the recent refactoring - so no more duplication for test / output parameters.

RazvanN7 commented:
@MoonlightSentinel is this ready for review?

MoonlightSentinel commented:
Yes
